frontend/src/__tests__/cypress/cypress/tests/e2e/pipelines/pipelines.cy.ts

```diff
@@ -0,0 +1,194 @@
+    const dspaSecretReplacements = {
+      DSPA_SECRET_NAME: dspaSecretName,
+      NAMESPACE: projectName,
+      AWS_ACCESS_KEY_ID: Buffer.from(AWS_PIPELINES_BUCKET.AWS_ACCESS_KEY_ID).toString('base64'),
+      AWS_SECRET_ACCESS_KEY: Buffer.from(AWS_PIPELINES_BUCKET.AWS_SECRET_ACCESS_KEY).toString(
+        'base64',
+      ),
+    };
+    cy.fixture('resources/yaml/dspa_secret.yml').then((yamlContent) => {
+      const modifiedYamlContent = replacePlaceholdersInYaml(yamlContent, dspaSecretReplacements);
+      const tempFilePath = 'cypress/temp_dspa_secret.yaml';
+      applyOpenShiftYaml(modifiedYamlContent, tempFilePath).then((result) => {
+        expect(result.code).to.eq(
+          0,
+          `ERROR applying YAML content
+                                      stdout: ${result.stdout}
+                                      stderr: ${result.stderr}`,
+        );
+      });
+    });
+
+    // Configure Pipeline server: Create DSPA
+    const dspaReplacements = {
+      DSPA_SECRET_NAME: dspaSecretName,
+      NAMESPACE: projectName,
+      AWS_S3_BUCKET: AWS_PIPELINES_BUCKET.BUCKET_NAME,
+    };
+    cy.fixture('resources/yaml/dspa.yml').then((yamlContent) => {
+      const modifiedYamlContent = replacePlaceholdersInYaml(yamlContent, dspaReplacements);
+      const tempFilePath = 'cypress/temp_dspa.yaml';
+      applyOpenShiftYaml(modifiedYamlContent, tempFilePath).then((result) => {
+        expect(result.code).to.eq(
+          0,
+          `ERROR applying YAML content
+                                      stdout: ${result.stdout}
+                                      stderr: ${result.stderr}`,
+        );
+      });
+    });
+  });
+
+  after(() => {
+    // Delete provisioned Project
+    deleteOpenShiftProject(projectName).then((result) => {
+      expect(result.code).to.eq(
+        0,
+        `ERROR deleting ${projectName} Project
+                                    stdout: ${result.stdout}
+                                    stderr: ${result.stderr}`,
+      );
+    });
+  });
+
+  it('An admin User can Import and Run a Pipeline', () => {
+    // Login as an admin
+    cy.visitWithLogin('/', ADMIN_USER);
+    cy.findByRole('banner', { name: 'page masthead' }).contains(ADMIN_USER.USERNAME);
+
+    /**
+     * Import Pipeline by URL from Project Details view
+     */
+    projectListPage.navigate();
+
+    // Check if "All available projects" is selected in the DS Projects view
+    projectListPage
+      .findProjectsTypeDropdown()
+      .invoke('text')
+      .then((text) => {
+        if (!text.includes(allAvailableProjectsText)) {
+          projectListPage.findProjectsTypeDropdown().click();
+          projectListPage.findProjectsTypeDropdownByText(allAvailableProjectsText).click();
+        }
+      });
+
+    // Open the project
+    projectListPage.findProjectLink(projectName).click();
+
+    // Increasing the timeout to ~2mins so the DSPA can be loaded
+    projectDetails.findImportPipelineButton(120000).click();
```

Comment:
We need to find a better way to avoid simply increasing wait times. 
If there's nothing in the UI to provide a proper status then perhaps we need a mechanism to also watch resources and react to error states.

Reply:
This is not a passive wait, is an active one. It's polling fot the element until it's available. And it's the same behaviour an end user will do (wait until the spinner is gone and the Import Pipeline button is available). 
If it reach the 2 minutes (now 3 minutes because sometimes the Pipeline Server takes a little bit longer to be available) it will fail. But if it's available in 2 seconds, the click will be performed after those 2 seconds

Reply:
I agree it's not just a `cy.wait`. What I'm getting at here is how do we know what is a long enough wait time to set for a test case.
We can keep increasing the timeout until it's "good enough". This is fine for the time being and I'm not requesting a change here today.

In general this is a bad approach but maybe all we got for now. If we can be more precise with watching events on a more timely basis, we can avoid having very long timeouts if we can expect a status event to occur within 10 seconds (as an example), then we watch for such events.

I want us to find a better way for our tests to be written to not have flakes because in future the timeout needs to be increased. Perfect example is you started with 2 mins but now increased to 3 mins.

Similarly, if we can find a way to create the test as to have wait for success or error states that could also alleviate some unnecessary wait times when a test is failing.

Reply:
I agree with you that we should not let the tests go on too long. In this case, the entire test takes ~3 minutes. Much of this time is spent on resource provisioning (deploying a DSP, and getting everything ready to deploy a pipeline). 
I also believe that tests should be atomic and deterministic, so that each test can be run independently (without having dependencies with another test). But that does not mean that each test should have its own specific resources. There are resources that will be provisioned once for all tests, other resources that will be provisioned for a complete suite, and others that will be provisioned for a single test. So if a single test is run, all granularity levels will need to be provisioned (which is not very efficient), but if we have a large test suite many of the resources will be provisioned only once, which increases efficiency.
For this case, I can imagine that in this suite other tests related to pipelines will be added to make use of these resources.